protocol/daemons/types/paginated_grpc_request.go

```diff
@@ -0,0 +1,64 @@
+package types
+
+import (
+	"context"
+	"github.com/cosmos/cosmos-sdk/types/query"
+	pricestypes "github.com/dydxprotocol/v4-chain/protocol/x/prices/types"
+)
+
+const (
+	// PaginatedRequestLimit is the maximum number of entries that can be returned in a paginated request
+	PaginatedRequestLimit = 10000
+)
+
+// AllPaginatedMarketParams returns all MarketParams from the prices module, paginating through the results and
+// returning a list of all the aggregated market-params.
+func AllPaginatedMarketParams(ctx context.Context, client pricestypes.QueryClient) ([]pricestypes.MarketParam, error) {
```

Comment:
nit: maybe we can move this function to daemons/pricefeed?

Reply:
This is method is used in both the pricefeed / slinky daemon. Hence I didn't want to introduce a dependency from one of those to the other. Happy to move this into pricefeed if this is not a concern

Reply:
Gotcha makses sense!